base/compiler/types.jl

```diff
@@ -45,6 +45,11 @@ The effects are composed of the following set of different properties:
 - `terminates::TriState`: this method is guaranteed to terminate
 - `nonoverlayed::Bool`: indicates that any methods that may be called within this method
   are not defined in an [overlayed method table](@ref OverlayMethodTable)
+- `notls::TriState`: this method does not access any state bound to the current
+  task and may thus be moved to a different task without changing observable
```

Comment:
Shall we add a clarification statement "except for the ptls and tid"?

Reply:
Well, it's allowed to access those for implementation-defined reasons, but if it access them to do something with other task-local state that's a problem obviously. Open to any suggestions for improved wording in that direction.